plugins/node/instrumentation-react-native-redux/src/middleware.ts

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { Dispatch, Middleware, Action, UnknownAction } from 'redux';
+import { TracerProvider, trace } from '@opentelemetry/api';
+import { PACKAGE_NAME, PACKAGE_VERSION } from './version';
+import { spanEnd, spanStart } from './utils/spanFactory';
+import { AppState } from 'react-native';
+
+const SPAN_NAME = {
+  action: 'redux-action',
+  store: 'redux-created-store',
+};
+
+interface MiddlewareConfig {
+  debug?: boolean;
+  allowLogging?: boolean;
+}
+
+const middleware = <RootState>(
+  provider: TracerProvider,
+  config?: MiddlewareConfig
+): Middleware<object, RootState> => {
+  /**
+   * happening just once, before the store is created
+   */
+
+  // NOTE
+  // create an span that start with the Store creation?
+  // should be ended when the store is destroyed (if possible to catch that moment)
+
+  // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+  // @ts-ignore
+  return () => {
+    /**
+     * happening just once, when the store is created
+     */
+    if (!provider) {
+      console.info('No TracerProvider found. Using global tracer instead.');
```

Comment:
could make configurable like the other PR

Reply:
are you talking about the console messages? if so, yes. to make it configurable it's under my plans. if you talk about the Provider, I think it's configured by the following couple of lines

Reply:
yep console messages

Reply:
done.